src/ScalarType.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Crell\Serde;
+
+enum ScalarType: string
```

Comment:
Problem: This doesn't allow for declaring an array of arrays.  Yet arrays aren't actually "scalars", so the enum name is not quite right.

I'm not sure if the backing values are useful.  Also, no blank line between the cases.

Reply:
Yes, no array of array implemented here as I mentioned in the issues as this is somehow wobbly IMHO and in such situation it should be mapped into another DTO... So I implemented scalars only.

Or do I miss something?

Reply:
I agree that array of arrays is usually wrong, but there are cases where it's valid.  I don't think we should forbid it.